pytorch_tabular/models/mixture_density/mdn.py

```diff
@@ -134,19 +136,14 @@ def generate_point_predictions(self, pi, sigma, mu, n_samples=None):
 
 class BaseMDN(BaseModel, metaclass=ABCMeta):
     def __init__(self, config: DictConfig, **kwargs):
-        assert config.task == "regression", "MDN is only implemented for Regression"
-        assert config.output_dim == 1, "MDN is not implemented for multi-targets"
+        assert config.task in ["regression", "ssl"], "MDN is only implemented for Regression"
```

Comment:
I think MDN is not relevant for SSL task. Only thing that is different in MDN is the specific MDN Head which estimates probabilistic parameters..

Reply:
Fixed, and I have done the same change also for Tabnet, because in this moment it's not useful use the ssl task since the head of the tabnet is doing almost nothing

Reply:
Can we also remove the SSL related code in the MDN models?

Reply:
I have removed the related code about ssn